consensus/hotstuff/pacemaker/pacemaker.go

```diff
@@ -27,15 +27,17 @@ import (
 //
 // Not concurrency safe.
 type ActivePaceMaker struct {
-	ctx                      context.Context
-	timeoutControl           *timeout.Controller
-	proposalDurationProvider ProposalDurationProvider
-	notifier                 hotstuff.Consumer
-	viewTracker              viewTracker
-	started                  bool
+	hotstuff.ProposalDurationProvider
```

Comment:
Is there a reason to couple `ActivePaceMaker` with `ProposalDurationProvider` after reviewing Jordan's PRs I think it makes sense to inject `ProposalDurationProvider` in `EventHandler` since it doesn't have any dependency on pacemaker itself.